gulpfile.js

```diff
@@ -92,43 +106,48 @@ var frontendDependencies = manifest.dependencies.frontend;
 var cssFilesPath = [];
 var jsFilesPath = [];
 var fontFolderPath = [];
-var cssBackgroundPath = [];
+var cssBackgroundFilepaths = [];
 var generatedTargetDir = path.join(
   'third_party', 'generated',
   isMinificationNeeded ? 'prod' : 'dev');
 var generatedCssTargetDir = path.join(generatedTargetDir, 'css');
 var generatedJsTargetDir = path.join(generatedTargetDir, 'js');
 
-for (var dependencyId in frontendDependencies) {
-  var dependency = frontendDependencies[dependencyId];
-  var dependencyDir = dependency.targetDirPrefix + dependency.version;
-  if (dependency.hasOwnProperty('bundle')) {
-    if (dependency.bundle.hasOwnProperty('css')) {
-      dependency.bundle.css.forEach(function(cssFiles) {
-        cssFilesPath.push(path.join(
-          'third_party', 'static', dependencyDir, cssFiles));
-      });
-    }
-    if (dependency.bundle.hasOwnProperty('js')) {
-      dependency.bundle.js.forEach(function(jsFiles) {
-        jsFilesPath.push(path.join(
-          'third_party', 'static', dependencyDir, jsFiles));
-      });
-    }
-    if (dependency.bundle.hasOwnProperty('fontsPath')) {
-      var fontPrefix = '*.{eot,woff2,ttf,woff,eof,svg}';
-      fontFolderPath.push(path.join('third_party', 'static', dependencyDir,
-        dependency.bundle.fontsPath, fontPrefix));
-    }
-    if (dependency.bundle.hasOwnProperty('cssBackgroundImage')) {
-      dependency.bundle.cssBackgroundImage.forEach(function(imagePath) {
-        cssBackgroundPath.push(path.join(
-          'third_party', 'static', dependencyDir, imagePath));
-      });
+
+gulp.task('collectFilesPath', function() {
```

Comment:
actually -- I'm looking at this again and thinking, filepaths of what? So, how about collectDependencyFilepaths?